kong/plugins/key-auth/handler.lua

```diff
@@ -193,11 +149,12 @@ local function do_authentication(conf)
 
   -- retrieve the consumer linked to this API key, to set appropriate headers
 
-  local consumer_cache_key = singletons.db.consumers:cache_key(credential.consumer_id)
-  local consumer, err = cache:get(consumer_cache_key, nil, load_consumer,
-                                  credential.consumer_id)
+  local consumer_cache_key = kong.db.consumers:cache_key(credential.consumer_id)
+  local consumer, err      = cache:get(consumer_cache_key, nil, load_consumer,
+                                       credential.consumer_id)
   if err then
-    return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
+    log_err(err)
```

Comment:
It seems like this pattern is repeated in multiple places? Any particular reason why?